speculation-rules.bs

```diff
@@ -627,6 +629,96 @@ A <dfn>prerender candidate</dfn> is a [=struct=] with the following [=struct/ite
   1. [=Assert=]: This step is not reached.
 </div>
 
+<h2 id="content-security-policy">Content Security Policy</h2>
+Inline script use is restricted by [[CSP]], but there are cases where a developer doesn't want to permit general inline scripts, but wants to permit inline speculation rules. So, we make the following change that extends [[CSP]] to support a new source keyword for inline speculation rules.
+
+<h3 id="content-security-policy-patches-directives">Directives</h3>
+In the [[CSP#framework-directives]], we define several associated algorithms, and an <dfn dfn-type="grammar" link-type="grammar">inline check</dfn> algorithm needs a patch to define how to choose the type argument.
+
+  3.  An <dfn for="directive" export>inline check</dfn>, which takes an {{Element}}, a
+      <dfn>`type`</dfn> string, a policy, and a source string as arguments,
+      and is executed during [[CSP#should-block-inline]] and during
+      [[CSP#should-block-navigation-request]] for `javascript:` requests. This
+      algorithm returns "`Allowed`" unless otherwise specified.
+      The <a>`type`</a> string is set to
+      "<dfn dfn-type="grammar" link-type="grammer">`script speculationrules`</dfn>"
+      if the {{Element}} is {{HTMLScriptElement}} and the {{HTMLScriptElement/type}}
+      attribute is "`speculationrules`", or set to
+      "<dfn dfn-type="grammar" link-type="grammer">`script attribute`</dfn>" for
+      JavaScript event handlers. Otherwise, it is set to 
+      "<dfn dfn-type="grammar" link-type="grammer">`script`</dfn>".
+
+<h3 id="content-security-policy-patches-source-lists">Source Lists</h3>
+In the [[CSP#framework-directive-source-list]], we change the following definition to have a new source, `inline-speculation-rules`.
+
+<pre dfn-type="grammar" link-type="grammar">
```

Comment:
Does this also need an extra attribute not to export dfn tags used inside the pre tag?
or should I also add noexport for each dfn tag?

Reply:
I think you can put the `noexport` on this container.

Reply:
Done!